File: 0739-daily-temperatures/0739-daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
         int n=temperatures.size();
         stack<pair<int,int>>temp;
          vector<int>ans(n);
        for(int i=0;i<n;i++)
        {
            while(temp.size()>0 && temp.top().first<temperatures[i])
            {
                ans[temp.top().second]=i-temp.top().second;
                temp.pop();
            }
            temp.push({temperatures[i],i});  
        }
        return ans;
    }
};

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        st = []
        n = len(temperatures)
        ans = [0] * n
        for i in range(n):
            while st and temperatures[i] > temperatures[st[-1]]:
                prev = st.pop()
                ans[prev] = i - prev
            st.append(i)
        return ans


        